demo/src/app/components/typeahead/overview/sections/customize-options-list/template.html

```diff
@@ -0,0 +1,41 @@
+<p>
+  You don't have to feed our typeahead with string items to make it work. It's actually better is you use your actual data model, since that is what gets committed when selecting an option.
+</p>
+
+<p>
+  But the requirement is to be able to <em>display</em> those options, be it in the list of the popup or inside the input field.
+</p>
+
+
+
+<h5>Providing a string</h5>
+
+<p>
+  The option item can be displayed from a string. In this case, use the property <code>resultFormatter</code> to pass a function which takes the option item (whose model is up to you) as input and returns a string as output.
```

Comment:
Describing this API in english, code is simpler, no ?

```ts
@Input() resultFormatter: (item: any) => string;
```

Reply:
Well, to know how to balance text vs code is not always easy. Sometimes text gives more information. In that case it seems equivalent. I'll think about it, good suggestion

Reply:
> Well, to know how to balance text vs code is not always easy

So was my point about "writing docs is hard